code/z_adventurezones/space_casino.dm

```diff
@@ -0,0 +1,371 @@
+				src.plays++
+				src.working = 1
+				src.icon_state = "slotsitem-on"
+
+				playsound(get_turf(src), "sound/machines/ding.ogg", 50, 1)
+				. = TRUE
+				ui_interact(usr, ui)
+				SPAWN_DBG(2.5 SECONDS) // why was this at ten seconds, christ
+					money_roll()
+					src.working = 0
+					src.icon_state = "slotsitem-off"
+
+			if("eject")
+				if(!src.scan)
+					return TRUE // jerks doing that "hide in a chute to glitch auto-update windows out" exploit caused a wall of runtime errors
+				usr.put_in_hand_or_eject(src.scan)
+				src.scan = null
+				src.working = FALSE
+				src.icon_state = "slotsitem-off" // just in case, some fucker broke it earlier
+				src.visible_message("<span class='subtle'><b>[src]</b> says, 'Thank you for playing!'</span>")
+				. = TRUE
+
+		src.add_fingerprint(usr)
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_SIGNAL, "machineUsed")
+
+	money_roll()
+		var/roll = rand(1,500)
+		var/exclamation = ""
+		var/win_sound = "sound/machines/ping.ogg"
+		var/obj/item/P = null
+
+		if (prob(src.uses))
+			src.emag_act(null, null) // bye bye!
+			return
+
+		if (roll <= 10) // rare tier, 2% chance
+			P = text2path(pick(raretier))
+			win_sound = "sound/misc/airraid_loop_short.ogg"
+			exclamation = "JACKPOT! "
+			src.uses += 20
+		else if (roll > 10 && roll <= 160) // half decent tier, 30% chance
+			P = text2path(pick(usefultier))
+			exclamation = "Big Winner! "
+		else // junk tier, 68% chance
+			P = text2path(pick(junktier))
+			exclamation = "Winner! "
+
+		if (P == null)
+			P = /obj/item/raw_material/rock
+		var/obj/item/prize = new P
+		prize.loc = src.loc
+		prize.layer += 0.1
+		src.visible_message("<span class='subtle'><b>[src]</b> says, '[exclamation][src.scan.registered] has won \an [prize.name]!'</span>")
+		playsound(get_turf(src), "[win_sound]", 55, 1)
+		src.working = 0
+		src.icon_state = "slotsitem-off"
+		src.uses++
+
+	emag_act(var/mob/user, var/obj/item/card/emag/E) // Freak out and die
+		src.icon_state = "slotsitem-malf"
+		playsound(get_turf(src), "sound/misc/klaxon.ogg", 55, 1)
+		src.visible_message("<span class='subtle'><b>[src]</b> says, 'WINNER! WINNER! JACKPOT! WINNER! JACKPOT! BIG WINNER! BIG WINNER!'</span>")
+		playsound(src.loc, "sound/impact_sounds/Metal_Clang_1.ogg", 60, 1, pitch = 1.2)
+		animate_shake(src,7,5,2)
+		sleep(3.5 SECONDS)
+
+		src.visible_message("<span class='subtle'><b>[src]</b> says, 'BIG WINNER! BIG WINNER!'</span>")
+		playsound(src.loc, "sound/impact_sounds/Metal_Clang_2.ogg", 60, 1, pitch = 0.8)
+		animate_shake(src,5,7,2)
+		sleep(1.5 SECONDS)
+
+		new/obj/decal/implo(src.loc)
+		playsound(src, 'sound/effects/suck.ogg', 60, 1)
+		if (src.scan)
+			src.scan.set_loc(src.loc)
+		qdel(src)
+
+// Barbuddy [my beloved]
+
+/obj/machinery/bot/barbuddy
+	name = "BarBuddy"
+	desc = "A little bartending robot!"
+	icon = 'icons/obj/bots/aibots.dmi'
+	icon_state = "robuddy1"
+	layer = 5.0 //TODO LAYER
+	density = 0
+	anchored = 0
+	bot_move_delay = FLOORBOT_MOVE_SPEED
```

Comment:
I'd make this its own unique define, again so if someone changes a floorbot's speed def, it doesnt suddenly make barbuddies zip around!